packages/analytics-js/src/components/configManager/ConfigManager.ts

```diff
@@ -96,14 +98,12 @@ class ConfigManager implements IConfigManager {
         this.logger,
       );
 
-      if (useServerSideCookies && dataServerUrl) {
-        const isValidUrl = isValidDataServerUrl(dataServerUrl, this.logger);
-        state.serverCookies.isEnabledServerSideCookies.value = isValidUrl;
-        state.serverCookies.dataServerUrl.value = isValidUrl
-          ? (removeTrailingSlashes(dataServerUrl) as string)
-          : undefined;
-      } else {
-        state.serverCookies.isEnabledServerSideCookies.value = useServerSideCookies as boolean;
+      if (useServerSideCookies) {
+        state.serverCookies.isEnabledServerSideCookies.value = useServerSideCookies;
+        state.serverCookies.dataServiceEndpoint.value = DEFAULT_DATA_SERVICE_ENDPOINT;
+        if (dataServiceEndpoint) {
+          state.serverCookies.dataServiceEndpoint.value = dataServiceEndpoint;
```

Comment:
We need to remove any leading slashes if present.

Reply:
Addressed

Reply:
We have to remove "leading" not trailing slashes.

Reply:
The sanitization should be similar to that of the trailing slashes utility but I'm going to let it slide for now.
But create a new ticket to address it later.